packages/oauth/src/state-stores/file-state-store.ts

```diff
@@ -0,0 +1,108 @@
+import { homedir } from 'os';
+import fs from 'fs';
+import path from 'path';
+import { randomUUID } from 'crypto';
+import { StateStore, StateObj } from './interface';
+import { InstallURLOptions } from '../install-url-options';
+import { InvalidStateError } from '../errors';
+
+export interface FileStateStoreArgs {
+  stateExpirationSeconds?: number;
+  baseDir?: string;
+}
+
+export class FileStateStore implements StateStore {
+  private baseDir: string;
+
+  private stateExpirationSeconds: number;
+
+  public constructor(args: FileStateStoreArgs) {
+    this.baseDir = args.baseDir !== undefined ?
+      args.baseDir :
+      `${homedir()}/.bolt-js-oauth-states`;
+    this.stateExpirationSeconds = args.stateExpirationSeconds !== undefined ?
+      args.stateExpirationSeconds :
+      600;
+  }
+
+  public async generateStateParam(
+    installOptions: InstallURLOptions,
+    now: Date,
+  ): Promise<string> {
+    const state = randomUUID();
```

Comment:
As JWT value is too long for file names, I chose UUID here. Also, the file content has simple JSON data structure. I think that there is no need to use JWT for server-side data.